src/Rating.jsx

```diff
@@ -141,10 +170,15 @@ var Rating = React.createClass({
       // indexes NaN or above the last full index.
       var percent = i - lastFullIndex === 0 ? index % 1 * 100 :
         i - lastFullIndex < 0 ? 100 : 0;
+
+	  var icon = (!this.state.selected &&
```

Comment:
Shouldn't `var icon` be outside the `for..loop`? I don't see any reason to calculate it on each iteration.